internal/packager/registry.go

```diff
@@ -38,39 +38,64 @@ func splitImageName(repotag string) (*imageComponents, error) {
 
 // Pull loads an app from a registry and returns the extracted dir name
 func Pull(repotag string, outputDir string) (string, error) {
-	payload, err := resto.PullConfigMulti(context.Background(), repotag, resto.RegistryOptions{})
+	imgRef, err := splitImageName(repotag)
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "origin %q is not a valid image name", repotag)
 	}
-	repoComps, err := splitImageName(repotag)
+	payload, err := resto.PullConfigMulti(context.Background(), repotag, resto.RegistryOptions{})
 	if err != nil {
 		return "", err
 	}
-	appDir := filepath.Join(outputDir, internal.DirNameFromAppName(repoComps.Name))
-	err = os.Mkdir(appDir, 0755)
-	if err != nil {
+	appDir := filepath.Join(outputDir, internal.DirNameFromAppName(imgRef.Name))
+	if err := os.MkdirAll(appDir, 0755); err != nil {
```

Comment:
I see this was changed to a `MkdirAll`. Why is this needed?

Reply:
This is a mistake, I was aligning the `fork` and `pull` code so they look as similar as possible so i can extract the duplication easier. I'll fix